lib/auth/init.go

```diff
@@ -789,6 +792,57 @@ func collectKeysInUse(cas ...types.CAKeySet) (keysInUse [][]byte) {
 	return keysInUse
 }
 
+// applyAuthroityConfig applies the latest keystore config to active keys updating
+// the stored CA if any changes occur.
+func applyAuthorityConfig(ctx context.Context, asrv *Server, ca types.CertAuthority) (types.CertAuthority, error) {
+	activeKeys := ca.GetActiveKeys()
+	var (
+		changed bool
+		err     error
+	)
+
+	apply := func(curr []byte) ([]byte, error) {
+		next, err := asrv.keyStore.ApplyConfig(ctx, curr)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		if !slices.Equal(curr, next) {
+			changed = true
+		}
+		return next, nil
+	}
+
+	for _, key := range activeKeys.SSH {
+		key.PrivateKey, err = apply(key.PrivateKey)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+	for _, key := range activeKeys.TLS {
+		key.Key, err = apply(key.Key)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+	for _, key := range activeKeys.JWT {
+		key.PrivateKey, err = apply(key.PrivateKey)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+	if !changed {
+		return ca, nil
+	}
+	if err := ca.SetActiveKeys(activeKeys); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	ca, err = asrv.UpdateCertAuthority(ctx, ca)
```

Comment:
I take it this is okay because it runs under the auth init lock so multiple auth servers shouldn't be racing to replicate the keys then update the cert authority yeah? could we just add a comment about that (i forgot about the auth init lock on first read)

Reply:
added the comment. Is it worth it to do a compare and swap here too?

Reply:
>I assume it's fine and that always having the primary region in the keyID isn't actually critical and it should get fixed the next time applyAuthorityConfig runs

correct.